pubsub/Readme.md

```diff
@@ -25,3 +25,44 @@ type PubSub interface {
 	Subscribe(req SubscribeRequest, handler func(msg *NewMessage) error) error
 }
 ```
+
+### TTL (or Time To Live)
+
+Time to live is implemented by default in Dapr. A publishing application can add the `expiration` attribute to a CloudEvent object and Dapr runtime will drop the message if the expiration is due at the subscriber's end, prior to invoking subscriber applications. The `expiration` attribute is Dapr specific and extends the CloudEvent spec. This is a convenience to subscriber applications, so Dapr does not expire messages from pub subs that do not have subscribers via Dapr.
+
+Alternatively, the publisher application can set the expiration by publishing with the `ttlInSeconds` metadata. Components that support TTL should parse this metadata attribute. Dapr runtime will also automatically populate the `expiration` attribute in the CloudEvent object if `ttlInSeconds` is present - in this case, Dapr will add 1 second to allow component implementations to expire the message before Dapr consumes it.
+
+As per the CloudEvent spec, `expiration` is formatted using RFC3339.
+
+Example:
+```json
+{
+	"specversion" : "1.0",
+	"type" : "com.github.pull.create",
+	"source" : "https://github.com/cloudevents/spec/pull",
+	"subject" : "123",
+	"id" : "A234-1234-1234",
+	"expiration" : "2018-04-06T17:31:00Z",
```

Comment:
To make this feature easier to understand, I am considering to rename `expiration` to `daprExpiration` and not document it as something users can set from application side (or SDK). So, expiration would always come from the `ttlInSeconds` metadata.

Reply:
Agreed, I don't think there is a good reason to have both expiration and ttlInSeconds exposed to the user. They are conceptually easy to get mixed up.

Reply:
I'm a little confused about why we'd use `daprExpiration` rather than `expiration` - and also who's setting it in this example - is this and example of the payload produced by a pub/sub component?

Reply:
@rynowak Applications can publish in two ways: send Dapr a CloudEvent (which, in this case, the application could also set the `expiration` attribute) or send the payload and let Dapr automatically wrap it into a CloudEvent. So, `expiration` can be set by Application. Dapr also supports (as part of this PR) the `ttlInSeconds` metadata, so that could also set the `expiration` attribute in Dapr.

Reply:
I kept `expiration` as the name. It is less verbose.

Reply:
Also, `expiration` is not dropped from the CloudEvent if the applications injects it. It is only set by Dapr via `ttlInSeconds` metadata.